### PR TITLE
class_isa representer option

### PR DESCRIPTION
**Pull request by pplu**
Hi,

  First commit is just basic tests for some existing representers. Second implements the class_isa representer option and adds tests for it

### CODE REVIEW

**Comment by perlpunk** [comment]
Thanks! I'll add some comments...

File: t/40.representers.t [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use Test::More;
+
+use YAML::PP;
+
+{
+  my $parser = YAML::PP->new;
```
Comment (perlpunk) [inline t/40.representers.t]: Could you change the indentation also to 4 spaces like in the module?

File: t/40.representers.t [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    o2 => (bless {}, 'Class2'),
+  };
+
+  my $yaml = $parser->dump_string($data);
+  # o1 serializes to Class1 because the first catchall says it's done
+  like($yaml, qr/o1: !Class1/, 'o1s\' gets caught only by the first class_matches, since it sets work as done');
+  like($yaml, qr/o2: !Class2/, 'o2s\' gets caught by the second class_matches');
+}
+
+package BaseClass {
```
Comment (perlpunk) [inline t/40.representers.t]: This package syntax is only supported starting with 5.14.

**Comment by pplu** [comment]
Done :dancer:

**Comment by perlpunk** [comment]
Thanks, merged!

**Pull request merged**